students/shibin_mathew/lesson3/mailroom.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+import sys
+
+
+def optionOne():
+    # Function to Send a Thank You Email
+    quit = False
+    while not quit:
+        print("Please input your full name")
+        name = input()
+        if name == "list":
+            print("list of names")
+            for names in donors:
+                print(names['name'])
+            print("Please select from the list or add a new name")
+            name = input()
+            if name != 'quit':
+                print("Please enter a donation amount")
+                donation = input()
+                if donation != 'quit':
+                    for names in donors:
+                        if names['name'] == name:
+                            names['donation'].append(float(donation))
+                    print(donors)  # Debug
+                    quit = True
+                else:
+                    quit = True
+            else:
+                quit = True
+        elif name == 'quit':
+            quit = True
+        else:
+            print("Please enter a donation amount")
+            donation = float(input())
+            if donation != 'quit':
+                donors.append({'name': name, 'donation': [donation]})
+                print(donors)  # Debug
+                quit = True
+            else:
+                quit = True
```

Comment:
consider refactoring above, there are too many if / else statements and too many places where quit is being kept tracked. remember that you can also use `break` statement to exit a loop.